src/platform/plugins/shared/discover/public/application/main/components/top_nav/use_top_nav_links.tsx

```diff
@@ -127,7 +146,15 @@ export const useTopNavLinks = ({
       }
 
       return items;
-    }, [discoverParams, state, services, defaultMenu, onOpenInspector, newSearchUrl]);
+    }, [
+      defaultMenu,
+      services,
+      onOpenInspector,
+      discoverParams,
+      state,
+      isEsqlMode,
+      currentDataView,
+    ]);
```

Comment:
Found https://github.com/elastic/kibana/issues/215676 while testing the PR but it also happens on main.